### ISSUE TITLE
Feature req: Buddy task

### ISSUE THREAD

**Issue by beltoft** [PokemonGoF/PokemonGo-Bot-Backup#10]
### Short Description

A new task to select a buddy pokemon
### Possible solution

a task named buddy where you could put in the name of the pokemon you would select as buddy.  
### How it would help others

it will give extra candy for the type of pokemon you chose as buddy for every km walked making it easier to evolve that type

**Comment by DeXtroTip** [comment]
I'm working on it, I'm running some tests before making a PR

**Comment by tohyf** [comment]
if a buddy is already selected would it require additional API calls to "walk" it ?

**Comment by DeXtroTip** [comment]
There is an api call to get  the reward, not sure if reward is got without it, maybe not

**Comment by tohyf** [comment]
Looks like the distance is being recorded down correctly, I guess the only question is whether the candy reward requires another API call

**Comment by DeXtroTip** [comment]
It might be like CollectLevelUpRewards or IncubateEggs, if you don't make the api call you don't get the rewards, I guess

**Comment by cowst** [comment]
That's the case, since there is another open issue exactly for that :)

**Comment by DeXtroTip** [comment]
Yeah, I've checked it already. Added check on task to get old rewards from current buddy.
Need to update docs/configuration_files.md before PR and the api also has a bug on buddy api calls, I had to fix it manually, we'll need to wait for them to merge the fix too before this task works.
